modules/caddyhttp/h2chandler.go

```diff
@@ -0,0 +1,80 @@
+package caddyhttp
+
+import (
+	"context"
+	"math/rand"
+	"net/http"
+	"net/textproto"
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/net/http/httpguts"
+)
+
+// h2chandler is a Handler which counts possible h2c upgrade requests
+type h2chandler struct {
```

Comment:
Shouldn't this be uppercase, as a separate word?
```suggestion
type h2cHandler struct {
```